### PR TITLE
#2072: hide add reaction when limit is reached

### PR DESCRIPTION
**Pull request by thebino**
fixes: #2072

hide/show add reaction chip based on the reaction count

[device-2022-11-06-183243.webm](https://user-images.githubusercontent.com/841681/200185881-ffe1cb1d-2832-47e5-a99b-14af62d9c362.webm)

### CODE REVIEW

File: app/src/main/java/com/keylesspalace/tusky/components/announcements/AnnouncementAdapter.kt [hunk 1]
```diff
@@ -73,6 +73,13 @@ class AnnouncementAdapter(
             return
         }
 
+        // hide button if announcement badge limit is already reached
+        if (item.reactions.size >= 8) {
+            addReactionChip.visibility = View.GONE
+        } else {
+            addReactionChip.visibility = View.VISIBLE
+        }
```
Comment (connyduck) [inline app/src/main/java/com/keylesspalace/tusky/components/announcements/AnnouncementAdapter.kt:81]: We have a neat extension function for this, you can write it as `addReactionChip.visible(item.reactions.size < 8)`

Comment (thebino) [reply app/src/main/java/com/keylesspalace/tusky/components/announcements/AnnouncementAdapter.kt]: ✅ adjusted

Comment (connyduck) [reply app/src/main/java/com/keylesspalace/tusky/components/announcements/AnnouncementAdapter.kt]: You got the condition the wrong way

Comment (thebino) [reply app/src/main/java/com/keylesspalace/tusky/components/announcements/AnnouncementAdapter.kt]: sorry, stupid me :)

**Pull request merged**